From 351fbd6208739d1a94fc11b2aa60d8a02d2f675a Mon Sep 17 00:00:00 2001
From: David Abdurachmanov <david.abdurachmanov@sifive.com>
Date: Mon, 13 Sep 2021 03:09:05 -0700
Subject: [PATCH 01/16] riscv: sifive: unleashed: support compressed images

Define kernel_comp_addr_r and kernel_comp_size to allow support for compressed
Image files for booti command. Note that we leave 128MiB for ramdisk and set
compressed kernel size for 64MiB.

Signed-off-by: David Abdurachmanov <david.abdurachmanov@sifive.com>
---
 include/configs/sifive-unleashed.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/configs/sifive-unleashed.h b/include/configs/sifive-unleashed.h
index 0d69d1c..5acce36 100644
--- a/include/configs/sifive-unleashed.h
+++ b/include/configs/sifive-unleashed.h
@@ -71,6 +71,8 @@
 	"script_size_f=0x1000\0" \
 	"pxefile_addr_r=0x88200000\0" \
 	"ramdisk_addr_r=0x88300000\0" \
+	"kernel_comp_addr_r=0x90300000\0" \
+	"kernel_comp_size=0x4000000\0" \
 	"type_guid_gpt_loader1=" TYPE_GUID_LOADER1 "\0" \
 	"type_guid_gpt_loader2=" TYPE_GUID_LOADER2 "\0" \
 	"type_guid_gpt_system=" TYPE_GUID_SYSTEM "\0" \
-- 
2.7.4

